tripal/src/TripalStorage/TripalStorageBase.php

```diff
@@ -207,23 +207,43 @@ public function getStoredValues() {
   }
 
   /**
-   * A simple helper function to clone a values array.
+   * A helper function to clone a values array.
    *
    * @param array $values
    *   An array of property values.
    */
   protected function cloneValues($values) {
     $copy = [];
     foreach ($values as $field_name => $deltas) {
+      $copy[$field_name] = [];
       foreach ($deltas as $delta => $keys) {
+        $copy[$field_name][$delta] = [];
         foreach ($keys as $key => $value) {
+          $copy[$field_name][$delta][$key] = [];
           $copy[$field_name][$delta][$key]['value'] = clone $value['value'];
         }
       }
     }
     return $copy;
   }
 
+  /**
+   * A helper function to add a new item for a field by cloning delta 0.
+   *
+   * @param array $values
+   *   An array of property values.
+   * @param string $field_name
+   *   The name of the field to addd an item to.
+   */
+  protected function addEmptyValuesItem(&$values, $field_name) {
+    $num_items = count($values[$field_name]);
+    $values[$field_name][$num_items] = [];
+    foreach ($values[$field_name][0] as $key => $value) {
+      $values[$field_name][$num_items][$key] = [];
+      $values[$field_name][$num_items][$key]['value'] = clone $value['value'];
```

Comment:
This line worries me because if this function is called after the value has been set for delta 0 then it will get cloned into other delta. This could cause bleeding of values between delta.

Perhaps we just set the value to empty as well 🤔 
```suggestion
      $values[$field_name][$num_items][$key]['value'] = clone $value['value'];
      $values[$field_name][$num_items][$key]['value']->setValue(NULL);
```

Reply:
Yeah, I tried that originally but it wouldn't work. I forget why.  My recollection is that each property has checks and wouldn't allow a null in all cases.  I can try again.

Reply:
I was wrong. This fix does work. I do remember having it this way at one point but it broke so I took out the `setValue()`.  But I must have confused myself on something.  It's working with it so I'm accepting.